userprofile/validators.py

```diff
@@ -0,0 +1,26 @@
+import functools
+
+import phonenumbers
+from iso6709 import Location
+from rest_framework.serializers import ValidationError
+
+
+def geographic_coordinate(value):
+    try:
+        Location(value)
+    except AttributeError:
+        raise ValidationError("Invalid ISO6709 geographic coordinate")
+
+
+def _phone_number(value, country):
+    try:
+        number = phonenumbers.parse(value, country)
+    except phonenumbers.NumberParseException as e:
+        raise ValidationError(str(e))
+    if not phonenumbers.is_possible_number(number):
+        raise ValidationError("Not a possible phone number")
+    if not phonenumbers.is_valid_number(number):
+        raise ValidationError("Not a valid phone number")
+
+
+za_phone_number = functools.partial(_phone_number, country="ZA")
```

Comment:
Does the phonenumber field that you used for the confirmed contact app do the same logic as this? If so, maybe we can replace all this custom code with that library? I think the important things are:
1. It takes user input and stores it in the database in international format. eg. User input on API: `0820001001`, value in db: `+27820001001`
2. It sends validation errors to the user for impossible, or invalid numbers, or numbers that we can't parse.

I see we do have a unit test for 1 (it takes `27820001001` and ensures it's stored in the db as `+27820001001`), but we don't have a unit test for 2

Reply:
I'll be looking into this right now, I think it should be possible to replace it

Reply:
Attempted to replace current custom validation with `django-phonenumber-field`. It does include the `phonenumbers` module but from what I can tell, does not provide the same functionality. I'm currently attempting to resolve migration issues.

Reply:
`userprofile_healthcheckuserprofile_msisdn_b48139f5_like` exists because of implied `unique=True` due to `primary_key=True` ([reference here](https://docs.djangoproject.com/en/3.1/ref/models/fields/#primary-key))

Reply:
(also: I tried to do an `unique=False` - it breaks the tests)

Reply:
Same goes for `userprofile_covid19triage_deduplication_id_key` - removing `unique=True` breaks tests. Breaking tests means it's possible to create duplicate instances - which is not intended. It could be possible to just remove those indexes via a custom migration - not sure how good of a solution that would be.

Reply:
For `userprofile_covid19triage_deduplication_id_d43bee39_like` and `userprofile_healthcheckuserprofile_msisdn_b48139f5_like`, what happens here is that if you create an index on a charfield in django, django always creates two indexes, one btree on the field, and another one on the field's `varchar_pattern_ops`, which is used for LIKE queries.

Now for these two fields, we're never going to run LIKE queries on them. And since these are going to be large tables, we don't want to have to pay the overhead of maintaining indexes that we'll never use. So what I usually do is add manual django migrations that delete those LIKE indexes.

For for example, we want the `userprofile_covid19triage_deduplication_id_key` index, but not the `userprofile_covid19triage_deduplication_id_d43bee39_like` index. Adding the `unique=True` creates both indexes, but we only want the one, so we should have the `unique=True` and the automated indexes, and then immediately afterwards delete the `_like` index. Same as we want the `userprofile_healthcheckuserprofile_pkey` index, but not the `userprofile_healthcheckuserprofile_msisdn_b48139f5_like` index.

Reply:
If `django-phonenumber-field` doesn't appear to have the same functionality, then leave it and keep the custom code. I was hoping it was a quick win, if not, then it's fine to have that custom code.

Reply:
Got it, will modify the migrations